### ISSUE TITLE
Thread controls in pytorch_backend is not handled correctly (due to wrong assumption)

### ISSUE THREAD

**Issue by erenboz** [triton-inference-server/server#7836]
**Description**

Parallelism thread controls introduced in this [PR](https://github.com/triton-inference-server/pytorch_backend/pull/125) assume these parameters can be changed per model or be set more than once. However as can be followed from [libtorch](https://github.com/pytorch/pytorch/blob/6d4cd3e5f27792d9c902cb8d12403cad9e9351bc/aten/src/ATen/ParallelThreadPoolNative.cpp#L50) they are meant to be done once before runtime initialization.

We did internally experience crashes during model updates due to `INTER_OP_THREAD_COUNT`, however steps to reproduce crash is not available at this time.

**Triton Information**
What version of Triton are you using?
2.50.0 / 24.09

Are you using the Triton container or did you build it yourself?
No

**To Reproduce**
Steps to reproduce the behavior.
Declare `INTER_OP_THREAD_COUNT` or `INTRA_OP_THREAD_COUNT` parameter in more than one model and load them.

Describe the models (framework, inputs, outputs), ideally include the model configuration file (if using an ensemble include the model configuration file for that as well).

we set it e.g.
```
parameters {
  key: "INTER_OP_THREAD_COUNT"
  value {
    string_value: "10"
  }
}
parameters {
  key: "INTRA_OP_THREAD_COUNT"
  value {
    string_value: "1"
  }
}
```

**Expected behavior**
AFAIK, there is no way to declare global parameters per backend so if such parameters declared differently for different pytorch models, it should result in no-op with warning message provided by backend and should not crash.

**Comment by erenboz** [comment]
FYI @yongbinfeng @tanmayv25.

**Comment by tanmayv25** [comment]
Thanks @erenboz  for investigating and opening this issue. If it is an initialization level setting, we can probably pass these thread count settings as server command line options( --backend-config) from the server. Similar to how we pass some settings to TF backend [here]((https://github.com/triton-inference-server/tensorflow_backend?tab=readme-ov-file#command-line-options)(--backend-config).

**Comment by erenboz** [comment]
> Thanks @erenboz for investigating and opening this issue. If it is an initialization level setting, we can probably pass these thread count settings as server command line options( --backend-config) from the server. Similar to how we pass some settings to TF backend [here](https://github.com/triton-inference-server/tensorflow_backend?tab=readme-ov-file#command-line-options).

Right, sounds indeed correct place to do so. We do not have an appropriate build environment to make a fix for it at the moment, anybody can feel free to pick it up.
